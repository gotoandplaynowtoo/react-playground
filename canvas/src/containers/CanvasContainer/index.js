import React, { Component } from 'react';
import Canvas from 'components/Canvas';

class CanvasContainer extends Component {

  constructor(props) {
    super(props);
    this.stage = null;
    this.ctx = null;
    this.animId = null;
    this.w = 0;
    this.h = 0;
  }

  enterFrame = () => {
    this.animId = requestAnimationFrame(this.enterFrame);
    const ctx = this.ctx;
    const x = Math.random() * this.w;
    const y = Math.random() * this.h;
    const size = Math.random() * 10;
    ctx.fillRect(x, y, size, size);
  }

  componentDidMount() {
    this.stage.width = this.w = window.innerWidth;
    this.stage.height = this.h = window.innerHeight;
    this.ctx = this.stage.getContext('2d');
    this.enterFrame();
  }
  
  componentWillUnmount() {
    cancelAnimationFrame(this.animId);
  }

  render() {
    return <Canvas innerRef={(comp) => this.stage = comp} />
  }

}

export default CanvasContainer;