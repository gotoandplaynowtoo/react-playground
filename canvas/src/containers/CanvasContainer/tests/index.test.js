import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CanvasContainer from 'containers/CanvasContainer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CanvasContainer/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(<CanvasContainer/>).toJSON();
  expect(tree).toMatchSnapshot();
});