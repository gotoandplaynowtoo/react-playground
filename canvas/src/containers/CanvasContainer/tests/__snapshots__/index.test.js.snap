// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<canvas
  className="sc-bdVaJa fpcbkF"
/>
`;
